dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -568,14 +568,20 @@ public String getUri() {
     return getOptions().valueOf(optionUri);
   }
 
+  @Nonnull
+  public String getHostOrDefault() {
+    return getHost(OPT_HOST_DEFAULT);
+  }
+
   @CheckForNull
-  public String getHost() {
+  public String getHostOrNull() {
     return getOptions().valueOf(optionHost);
   }
 
   @Nonnull
   public String getHost(@Nonnull String defaultHost) {
-    return MoreObjects.firstNonNull(getHost(), defaultHost);
+    String nullableHost = getHostOrNull();
+    return nullableHost != null ? nullableHost : OPT_HOST_DEFAULT;
```

Comment:
I would suggest keeping the previous implementation.

1. The `nullableHost` should be called `host`, because the nullability is implied in the type.
2. After changing to:

```
String host = getHost();
return host != null ? host : OPT_HOST_DEFAULT;
```

you can then inline it to: `return MoreObjects.firstNonNull(getHost(), OPT_HOST_DEFAULT);`

3. The `defaultHost` parameter is not used, therefore further simplification leads to the original code:

```
return MoreObjects.firstNonNull(getHost(), defaultHost);
```

Reply:
Done